src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+export interface Figure {
+  getArea: () => number;
+  shape: string;
+  color: string;
+}
+
+export class Triangle implements Figure {
+  public shape: string;
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: string;
 
-export class Triangle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
 
-export class Circle implements Figure {}
+    if (radius <= 0) {
+      throw new Error('your error message');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * Math.pow(this.radius, 2);
+
+    return Math.round(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape: string;
+
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign `this.shape` in the constructor because TypeScript does this automatically when you use access modifiers in the constructor parameters.